great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -867,9 +886,14 @@ def get_batch_data_and_markers(
                 source_schema_name=source_schema_name,
             )
         elif isinstance(batch_spec, SqlAlchemyDatasourceBatchSpec):
-            selectable: Select = self._build_selectable_from_batch_spec(
-                batch_spec=batch_spec
-            )
+            if self.engine.dialect.name.lower() == "oracle":
+                selectable: str = self._build_selectable_from_batch_spec(
+                    batch_spec=batch_spec
+                )
+            else:
+                selectable: Select = self._build_selectable_from_batch_spec(
+                    batch_spec=batch_spec
+                )
 
```

Comment:
Thank you very much for this too.. the type hinting reveals a subtlety that could be easily overlooked :)